test cases/common/229 persubproject options/meson.build

```diff
@@ -1,10 +1,24 @@
-project('persubproject options', 'c', default_options : ['default_library=both'])
+project('persubproject options', 'c', 'cpp',
+  default_options : [
+    'default_library=both',
+    'c_args=-DPARENT',
+    'cpp_std=c++98'
+  ]
+)
 
 assert(get_option('default_library') == 'both', 'Parent default_library should be "both"')
+assert(get_option('c_args') == ['-DPARENT'], 'Parent c_args are wrong')
 
 # Check it build both by calling a method only both_libraries target implement
 lib = library('lib1', 'foo.c')
 lib.get_static_lib()
 
-subproject('sub1')
-subproject('sub2', default_options : ['default_library=static'])
+sub1 = subproject('sub1')
+sub2 = subproject('sub2',
+  default_options : [
+    'default_library=static',
+    'c_args=-DSUB2_FROM_PARENT',
+  ]
+)
+
+test('mixing-cpp-version', executable('test1', 'main.cpp', link_with : sub2.get_variable('libcpp11')))
```

Comment:
There are no ABI compatibility guarantees for c++98 and c++11 AFAIK. There *are* guarantees for c++11, c++14 and c++17 though.
I.e. it's safe to mix c++11, c++14 and c++17, but not with c++03/c++98.